.github/workflows/container.yaml

```
@@ -0,0 +1,85 @@
+name: Container Image
+
+on:
+  # Event triggered upon completion of the `release.yaml` workflow
+  workflow_run:
+    workflows: ['Release']
+    types:
+      - completed
+
+  # Manual trigger
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to build (e.g., 1.42.4)'
+        required: true
+        type: string
+
+  workflow_call:
+    inputs:
+      version:
+        description: 'Version to build (e.g., 1.42.4)'
+        required: true
+        type: string
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  container:
+    if: ${{ github.event.workflow_run.conclusion == 'success' || inputs.version }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: 'Set up QEMU'
+      uses: docker/setup-qemu-action@v3
+      with:
+        platforms: arm64
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Log in to Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Extract version
+      id: version
+      run: |
+        VERSION="${{ inputs.version }}"
+        echo "version=${VERSION}" >> $GITHUB_OUTPUT
+        echo "Building container for version: ${VERSION}"
+
+    - name: Extract metadata
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        # NOTE: The `semver` tag type below with the default `flavor.latest=auto` action setting,
+        # will implicitly append a `latest` tag to publish:
+        # https://github.com/docker/metadata-action/issues/461#issuecomment-2680849083
+        tags: |
+          type=semver,pattern={{version}},value=${{ steps.version.outputs.version }}
+          type=semver,pattern={{major}}.{{minor}},value=${{ steps.version.outputs.version }}
+          type=semver,pattern={{major}},value=${{ steps.version.outputs.version }}
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v6
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64
+        push: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        build-args: |
+          JUST_VERSION=${{ steps.version.outputs.version }}
```

With your changes to drop `on.release`, since you only have `inputs.version` to source the version from now, just use that directly?

```suggestion
    - name: Prepare image metadata
      id: metadata
      uses: docker/metadata-action@v5
      with:
        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
        # NOTE: The `semver` tag type below (with the default `flavor.latest=auto` action setting),
        # will implicitly append a `latest` tag to publish:
        # https://github.com/docker/metadata-action/issues/461#issuecomment-2680849083
        tags: |
          type=semver,pattern={{major}},value=${{ inputs.version }}
          type=semver,pattern={{major}}.{{minor}},value=${{ inputs.version }}
          type=semver,pattern={{major}}.{{minor}}.{{patch}},value=${{ inputs.version }}

    - name: Build and push Docker image
      uses: docker/build-push-action@v6
      with:
        context: .
        platforms: linux/amd64,linux/arm64
        push: true
        tags: ${{ steps.metadata.outputs.tags }}
        labels: ${{ steps.metadata.outputs.labels }}
        build-args: |
          JUST_VERSION=${{ inputs.version }}
```